api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1791,6 +1792,70 @@ type AzurePlatformSpec struct {
 	// +immutable
 	// +required
 	SecurityGroupID string `json:"securityGroupID,omitempty"`
+
+	// IngressClientID is the client ID appropriate for the Ingress Operator.
+	//
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="IngressClientID is immutable"
+	// +kubebuilder:validation:Required
+	// +immutable
+	// +required
+	IngressClientID string `json:"ingressClientID"`
```

Comment:
All of these we should be able to get super easily by using `ccoctl` from the release payload, but the node-pool management and CPO are going to be something we need to figure out and encode into `CredentialsRequests` ourselves.